packages/flutter/lib/src/rendering/editable_line.dart

```diff
@@ -147,6 +153,31 @@ class RenderEditableLine extends RenderBox {
 
   bool hitTestSelf(Point position) => true;
 
+  TapGestureRecognizer _tap;
+  void handleEvent(PointerEvent event, BoxHitTestEntry entry) {
+    if (event is PointerDownEvent && onSelectionChanged != null)
+      _tap.addPointer(event);
+  }
+
+  Point _lastTapDownPosition;
+  void _handleTapDown(Point globalPosition) {
+    _lastTapDownPosition = globalPosition;
+  }
+
+  void _handleTap() {
+    assert(_lastTapDownPosition != null);
+    final Point global = _lastTapDownPosition;
+    _lastTapDownPosition = null;
+    if (onSelectionChanged != null) {
+      TextPosition position = _textPainter.getPositionForOffset(globalToLocal(global).toOffset());
+      onSelectionChanged(new TextSelection.fromPosition(position));
```

Comment:
`onSelectionChanged` is a bit misleading. The selection property doesn't change. It's more like `onSelectionChangeRequest` or something like that. It took me a while to understand the ownership model for the selection.